backend/routes/upload_reservation_routes.py

```
@@ -133,39 +243,57 @@ def upload_file():
-                if not filepath:
-                    raise BadRequest(f"Failed to save {key}")
-                files[key] = filename
-
-                # Validate document text
-                is_valid, text = validate_document(filepath)
-                validation_results[key] = {"valid": is_valid, "extracted_text": text}
-                if not is_valid:
-                    raise BadRequest(f"Invalid document for {key}")
+            file_error = _validate_upload_file(file, key)
+            if file_error:
+                raise BadRequest(file_error)
```

Ruff (0.15.1)

[warning] Abstract `raise` to an inner function

(TRY301)